app/services/history-sync.js

```diff
@@ -282,7 +282,8 @@ export default Ember.Service.extend({
     const session = visit.get('session.content');
     this.get('urlsToDelete').addObject(visit.get('url'));
     visit.unloadRecord();
-    if (session.get('visits.length') === 0) {
+    if (session.get('visits.length') === 0 ||
```

Comment:
please put it on a session model in a form of computed property

Reply:
made it in this PR:  #11